controllers/comment-controllers.js

```diff
@@ -0,0 +1,43 @@
+const { deleteComment, addComment } = require("../models/comment-models");
+const {
+  checkCommentIdExists,
+  checkIdExists,
+  checkUserNameExist,
+} = require("../models/article-parameters-exists-models");
+
+const postComment = (request, response, next) => {
+  const { body } = request.body;
+  const { username } = request.body;
+  const { article_id } = request.params;
+  if (body === undefined || username === undefined) {
+    response.status(400).send({ msg: "missing required field" });
+  }
+
+  Promise.all([checkIdExists(article_id), checkUserNameExist(username)])
+    .then(() => {
+      return addComment(username, body, article_id);
+    })
+    .then((comments) => {
+      response.status(201).send({ comments });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+const removeComment = (request, response, next) => {
+  const { comment_id } = request.params;
+
+  checkCommentIdExists(comment_id)
```

Comment:
+1 for abstracting logic but this isn't strictly necessary as we will get an empty row back for this query which we could simply reject here

Reply:
I chose to do this just to stick with the format for previous. Can change if need be. Thank you for the feedback :) sorry about the pull request, I think something with my merge issues yesterday may have caused some issues